cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/NodeBuilder.kt

```diff
@@ -269,3 +270,81 @@ fun <T : Node, S> T.codeAndLocationFrom(frontend: LanguageFrontend<S, *>, rawNod
 
     return this
 }
+
+/**
+ * This function allows the setting of a node's code and location region as the code and location of
+ * its children. Sometimes, when we translate a parent node in the language-specific AST with its
+ * children into the CPG AST, we have to set a specific intermediate Node between, that has no
+ * language-specific AST that can give it a proper code and location.
+ *
+ * While the location of the node is determined by the start and end of the child locations, the
+ * code is extracted from the parent node to catch separators and auxiliary syntactic elements that
+ * are between the child nodes.
+ *
+ * @param frontend Used to invoke language specific code and location generation
+ * @param parentNode Used to extract the code for this node
+ */
+fun <T : Node, S> T.codeAndLocationFromChildren(
+    frontend: LanguageFrontend<S, *>,
+    parentNode: S
+): T {
+    var first: Node? = null
+    var last: Node? = null
+
+    // Search through all children to find the first and last node based on region startLine and
+    // startColumn
+    var worklist: MutableList<Node> = this.astChildren.toMutableList()
+    while (worklist.isNotEmpty()) {
+        var current = worklist.removeFirst()
+        if (current.location?.region == null || current.location?.region == Region()) {
+            // If the node has no location we use the same search on his children again
+            worklist.addAll(current.astChildren)
```

Comment:
Shoudn't you add the ast children in any case to traverse all the children? This no only traverses if the children's node does not have a location?

Reply:
No, because a child node should not have a location outside of the parent location, it is intentional to add the children only if the parent has no location.

Reply:
Ah, that makes sense.